sui_core/src/authority/authority_store.rs

```diff
@@ -516,21 +535,23 @@ impl<const ALL_OBJ_VER: bool> SuiDataStore<ALL_OBJ_VER> {
         &self,
         owned_input_objects: &[ObjectRef],
         tx_digest: TransactionDigest,
-        signed_transaction: SignedTransaction,
+        transaction: Either<Transaction, SignedTransaction>,
```

Comment:
Here I would define an enum, instead of using Either.

Reply:
More strategically: I think that we could in all cases only store one of transaction, signed transaction, or (certificate, effects) since they all contain the transaction. Maybe we should have 1 table TxDigest -> FormsOfTransaction that stores one for each tx digest?

Reply:
Good point. Just sent https://github.com/MystenLabs/sui/pull/1055 for this